mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -300,7 +300,7 @@ class LinearRegressionModel private[ml] (
   private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
     val t = udf { features: Vector => predict(features) }
     val predictionAndObservations = dataset
-      .select(col($(labelCol)), t(col($(featuresCol))).as($(predictionCol)))
+      .withColumn($(predictionCol), t(col($(featuresCol))))
```

Comment:
This should call transform(dataset) and store the output dataset, which can include probabilities, etc.